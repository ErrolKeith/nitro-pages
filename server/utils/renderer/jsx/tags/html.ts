import { makeJsx } from "./makeJsx";

export interface HtmlConfig {
  props: HtmlProps;
  children: React.ReactElement[];
}

export interface HtmlProps {
  className?: string;
  ["data-site-id"]: string;
}

export function makeHtml(config: HtmlConfig) {
  return makeJsx(`html`, config.props, config.children);
}
