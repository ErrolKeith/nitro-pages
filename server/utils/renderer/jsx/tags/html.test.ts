import { it, expect } from "vitest";
import { makeHtml, HtmlConfig } from "./html";

it("can make an html tag", () => {
  const htmlTestConfig: HtmlConfig = {
    props: {
      ["data-site-id"]: "1234235",
    },
    children: [],
  };
  const htmlTest = makeHtml(htmlTestConfig);
  expect(htmlTest).toMatchSnapshot();
});
