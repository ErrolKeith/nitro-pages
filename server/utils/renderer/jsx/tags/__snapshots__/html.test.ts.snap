// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can make an html tag 1`] = `
<html
  data-site-id="1234235"
/>
`;
